File: src/java/org/javaturk/dp/ch04/abstractFactory/gui/sol3/WinFactory.java
package org.javaturk.dp.ch04.abstractFactory.gui.sol3;

public class WinFactory implements GUIFactory {
    public Component createButton() {
        return new WinButton();
    }

	@Override
	public Component createList() {
		return new WinList();
	}

	@Override
	public Component createTable() {
		return new WinTable();
	}
}

File: src/typescript/com/gokselkucuksahin/dp/creational-patterns/abstract-factory/gui/sol3/win-factory.ts
import {GUIFactory} from "../sol2/gui-factory";
import {Component} from "../sol1/component";
import {WinButton} from "./win-button";
import {WinList} from "./win-list";
import {WinTable} from "./win-table";

export class WinFactory implements GUIFactory {
  createButton(): Component {
    return new WinButton();
  }

  createList(): Component {
    return new WinList();
  }

  createTable(): Component {
    return new WinTable();
  }
}